heat/core/operations.py

```diff
@@ -146,15 +147,17 @@ def allclose(x, y, rtol=1e-05, atol=1e-08, equal_nan=False):
 
     """
     if not isinstance(x, tensor.tensor):
-        raise TypeError('Expected x to be a ht.tensor, but was {}'.format(type(x)))
+        raise TypeError(
+            'Expected x to be a ht.tensor, but was {}'.format(type(x)))
 
     if not isinstance(y, tensor.tensor):
-        raise TypeError('Expected y to be a ht.tensor, but was {}'.format(type(y)))
+        raise TypeError(
+            'Expected y to be a ht.tensor, but was {}'.format(type(y)))
 
     return torch.allclose(x._tensor__array, y._tensor__array, rtol, atol, equal_nan)
 
 
-def argmin(x, axis=None, out=None):
+def argmin(x, axis=None, keepdim=False, out=None):
```

Comment:
keepdim is assumed to be auto-false if not set, so the parameter can be removed

Reply:
Hey Markus, I don't get it. If I remove the parameter from the definition, I can't use it in the __reduce_op call later.

Reply:
My point being: keepdim should not appear in the function signature of these calls here as they are also not part of the numpy API (see here: https://docs.scipy.org/doc/numpy/reference/generated/numpy.argmin.html)

Beyond that, most (all?) calls in PyTorch assume keepdim=False if keepdim is not explicitly set. Practically for implementation reasons you could do the following:

`
def argmin(x, axis=None):
    return __reduce_op(..., keepdim=False)

def __reduce_op(x, local_func, **kwargs):
    # access kwargs explicitly
    print(kwargs['keepdim'])
    # or even
    local_func(**kwargs) # pass whatever is in kwargs onward to local_func
`

Does that make sense or am I confused?

Reply:
OK fixed it.